project2-beat-that/script.js

```diff
@@ -1,4 +1,266 @@
+      Player 1 score: ${player1Score}`;
+    }
+    return `it's a draw! <br><br>
+    Player 1 score: ${player1Score} <br><br>
+    Player 2 score: ${player2Score}`;
+  }
 };
+
+/* need help: In this version, I tried incorporating player 1 and 2 playerScore logic into ONE function (getPlayerScore), but why is player 1 score not shown in results? */
+
+// var gameMode = "player1diceroll";
+
+// //initialize array to store 2 dices rolled.
+// var dicesRolled = [];
+// // var message = " ";
+
+// //initialize first player.
+// var currentPlayer = 1;
+
+// // Player 1 and Player 2 score.
+// var player1Score = 0;
+// var player2Score = 0;
+
+// // Temporary storage of dice numbers
+// var player1CurrentDiceRoll = [];
+// var player1CurrentDiceRoll = [];
+
+// //randomly generate a number  from 1 to 6
+// var generateRandomNumber = function () {
+//   return Math.floor(Math.random() * 6) + 1;
+// };
+
+// //push the dice number of each of 2 rolls into the array
+// var get2DiceRolls = function () {
+//   dicesRolled = [];
+//   for (var i = 0; i < 2; i += 1) {
+//     dicesRolled.push(generateRandomNumber());
+//   }
+//   return dicesRolled;
+// };
+
+// var getPlayerScore = function (userInput, playerScore, dicesRolled) {
+//   let firstDice = dicesRolled[0];
+//   let secondDice = dicesRolled[1];
+//   if (userInput === 1) {
+//     playerScore = Number(String(firstDice) + String(secondDice));
+//     if (currentPlayer == 1) {
+//       player1Score = playerScore;
+//     } else {
+//       player2Score = playerScore;
+//     }
+//     return `Based on the dice you've rolled, the sum of your score is ${playerScore}. <br><br> Click submit for the next instruction.`;
+//   }
+//   if (userInput === 2) {
+//     playerScore = Number(String(secondDice) + String(firstDice));
+//     //separate out accordingly.
+//     if (currentPlayer == 1) {
+//       player1Score = playerScore;
+//     } else {
+//       player2Score = playerScore;
+//     }
+//     return `Based on the dice you've rolled, the sum of your score is ${playerScore}. <br><br> Click submit for the next instruction.`;
+//   }
+// };
+
+// var main = function (input) {
+//   input = Number(input);
+//   // var gameMode;
+//   if (gameMode === "player1diceroll") {
+//     dicesRolled = get2DiceRolls();
+//     gameMode = "player1diceorder";
+//     return `Welcome to the game, player ${currentPlayer}. <br><br>
+//     You rolled 2 dices. <br>
+//     Dice 1: ${dicesRolled[0]}. <br><br>
+//     Dice 2: ${dicesRolled[1]}. <br><br>
+//     Now, enter either 1 or 2 to define the order of your dice roll. This program will then show you your score.
+//     `;
+//   }
+
+//   if (gameMode === "player1diceorder") {
+//     if (input != 1 && input != 2) {
+//       return `Please only enter the number 1 or 2.`;
+//     }
+//     gameMode = "player2turn";
+//     currentPlayer = 2;
+//     //show return current number player 1 chose.
+//     return getPlayerScore(input, player1Score);
```

Comment:
i think the issue is somewhere here

1. getPlayerScore takes in three parameters as per your definition above, but you are only passing in two
2. also here you want getPlayerScore to calc the score for player 1, but you are setting currentPlayer to 2 before that even happens

getPlayerScore depends on currentPlayer to determine which variable (player1Score or player2Score) to store the score in